x-pack/plugins/ui_actions_enhanced/server/plugin.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { identity } from 'lodash';
+import { CoreSetup, Plugin, SavedObjectReference } from '../../../../src/core/server';
+import { EmbeddableSetup } from '../../../../src/plugins/embeddable/server';
+import { dynamicActionEnhancement } from './dynamic_action_enhancement';
+import {
+  ActionFactoryRegistry,
+  SerializedEvent,
+  ActionFactoryDefinition,
+  DynamicActionsState,
+} from './types';
+import { PersistableState } from '../../../../src/plugins/kibana_utils/common';
+
+export interface SetupContract {
+  registerActionFactory: any;
+}
+
+export type StartContract = PersistableState<DynamicActionsState>;
+
+interface SetupDependencies {
+  embeddable: EmbeddableSetup; // Embeddable are needed because they register basic triggers/actions.
+}
+
+export class AdvancedUiActionsPublicPlugin
+  implements Plugin<SetupContract, StartContract, SetupDependencies> {
+  protected readonly actionFactories: ActionFactoryRegistry = new Map();
+
+  constructor() {}
+
+  public setup(core: CoreSetup, { embeddable }: SetupDependencies) {
+    // this.enhancements = new UiActionsServiceEnhancements({
+    //   getLicense: () => this.getLicenseInfo(),
+    //   featureUsageSetup: licensing.featureUsage,
+    //   getFeatureUsageStart: () => startServices().plugins.licensing.featureUsage,
+    // });
+    embeddable.registerEnhancement(dynamicActionEnhancement(this));
+
+    return {
+      registerActionFactory: this.registerActionFactory,
+    };
+  }
+
+  public start() {
+    return {
```

Comment:
Why is this exposed? Looks like it is enough to register it with the embeddable plugin

Reply:
even if there is no use case for it at the moment, we want to allow using user configured ui actions outside of dashboard/embeddable